### PR TITLE
Updates mobile automation setup docs for taskgraph

### PR DESCRIPTION
**Pull request by mitchhentges**

### CODE REVIEW

File: procedures/mobile_automation_setup.rst [hunk 1]
```diff
@@ -32,100 +26,62 @@ administrating them in the live Taskcluster website.
...
+    to decide what tasks to run, we can take a ``.taskcluster.yml`` from
+    a similarly-configured repository, like `fenix
+    example <https://github.com/mozilla-mobile/fenix/blob/master/.taskcluster.yml>`__)
+
+    -  Update repository and treeherder references to refer to your project,
+       rather than ``fenix``.
+
+4.  Implement ``taskgraph`` configuration for the repository. See the
+    `Fenix configuration <https://github.com/mozilla-mobile/fenix/tree/master/taskcluster>`__.
+    You'll need to implement the following parts:
```
Comment (JohanLorenzo) [inline procedures/mobile_automation_setup.rst]: Followup for me: Link to the blog post that gives an introduction to transforms, loaders and all.

File: procedures/mobile_automation_setup.rst [hunk 2]
```diff
@@ -32,100 +26,62 @@ administrating them in the live Taskcluster website.
...
+    -  Define tasks in YAML in ``taskcluster/ci/``
+    -  Define transforms in ``taskcluster/$project_taskgraph/transforms/`` which operate
+       on the tasks defined in the YAML
+    -  Define any custom loaders in ``taskcluster/$project_taskgraph/loader/`` (this is
+       useful in cases like needing to generate a dynamic number of tasks based on an
+       external source, like ``gradle`` or a ``.buildconfig.yml`` file)
+    -  Define ``Dockerfiles`` in ``taskcluster/docker/``
+
+5.  If you have schedule-based automation, create cron hook
+    (copy from the `fenix hook <https://tools.taskcluster.net/hooks/project-releng/cron-task-mozilla-mobile-fenix>`__)
```
Comment (JohanLorenzo) [inline procedures/mobile_automation_setup.rst]: Nit: No need 😃  `ci-admin` generates it for you

File: procedures/mobile_automation_setup.rst [hunk 3]
```diff
@@ -32,100 +26,62 @@ administrating them in the live Taskcluster website.
...
+
+    -  You'll need to update the references from ``fenix``
+    -  Additionally, define a ``.cron.yml``, like `fenix <https://github.com/mozilla-mobile/fenix/blob/master/.cron.yml>`__
+    -  Create some hooks to force-trigger some cron tasks for testing. For example, `fenix has one for forcing nightlies
```
Comment (JohanLorenzo) [inline procedures/mobile_automation_setup.rst]: Nit: Same here, `ci-admin` has got the powa!

File: procedures/mobile_automation_setup.rst [hunk 4]
```diff
@@ -32,100 +26,62 @@ administrating them in the live Taskcluster website.
...
+        2. Set up a ``virtualenv`` and install dependencies
+
+    1. ``hg clone https://hg.mozilla.org/ci/ci-configuration/``
+    2. Update ``projects.yml`` and ``grants.yml`` to add permissions for ``$project``
+    3. Submit your patch for review with `moz-phab <https://github.com/mozilla-conduit/review>`__
+    4. Once it's landed, update to the new revision and apply it
+
+        1. ``ci-admin diff --environment=production``
+        2. If there's no surprises in the diff, apply it: ``ci-admin apply --environment=production``
+        3. If the diff has unexpected contents, you can adjust the ``diff`` and ``apply`` operations with the ``--grep`` flag:
```
Comment (JohanLorenzo) [inline procedures/mobile_automation_setup.rst]: Nit: Let's explain what's unexpected in this context, and why it's okay to see that. For example: mozilla-{central,beta,release} hooks are regularly updated by ci-admin and they may show up in the diff until someones runs it.

File: procedures/mobile_automation_setup.rst [hunk 5]
```diff
@@ -153,7 +109,7 @@ administrating them in the live Taskcluster website.
              first)
 
 10. Update configuration in
-    ```build-puppet`` <https://github.com/mozilla-releng/build-puppet/>`__
+    `build-puppet <https://github.com/mozilla-releng/build-puppet/>`__
```
Comment (JohanLorenzo) [inline procedures/mobile_automation_setup.rst]: Followup: Soon this whole puppet section will go away 😃

File: procedures/mobile_automation_setup.rst (hunk 2 above)
Comment (mitchhentges) [reply procedures/mobile_automation_setup.rst]: How do you generate the cron hook with `ci-admin`? I can add it to the docs here

File: procedures/mobile_automation_setup.rst (hunk 4 above)
Comment (mitchhentges) [reply procedures/mobile_automation_setup.rst]: Sure, I defined "unexpected" as changes in the diff that are unrelated to the hooks/permissions that were created earlier in the workflow :)

File: procedures/mobile_automation_setup.rst (hunk 2 above)
Comment (JohanLorenzo) [reply procedures/mobile_automation_setup.rst]: Thanks to these 2 lines (if I remember correctly):

* https://hg.mozilla.org/ci/ci-configuration/file/6e2e5169451be654cbb8a6962fca9327250d57a3/projects.yml#l483
* https://hg.mozilla.org/ci/ci-configuration/file/6e2e5169451be654cbb8a6962fca9327250d57a3/projects.yml#l485

Comment (mitchhentges) [reply procedures/mobile_automation_setup.rst]: Thanks!

**Pull request merged**